core/domain/summary_services.py

```diff
@@ -163,6 +163,46 @@ def get_displayable_collection_summary_dicts_matching_ids(collection_ids):
     return _get_displayable_collection_summary_dicts(collection_summaries)
 
 
+def get_collection_node_exploration_dicts(
+        exploration_ids, editor_user_id=None):
+    """Given a list of exploration ids, optionally filters the list for
+    explorations that are currently non-private and not deleted, and returns a
+    list of dicts of the corresponding exploration summaries for collection
+    node search.
+
+    Args:
+        exploration_ids: str. The list of exploration ids for fetching their
+            metadata dicts.
+        editor_user_id: str or None. The user_id of the editor to check for the
+            rights for an exploration.
+    Returns:
+        List of metadata dicts for the set of exploration ids.
+    """
+    exploration_summaries = (
+        exp_services.get_exploration_summaries_matching_ids(exploration_ids))
+
+    filtered_exploration_summaries = []
+    for exploration_summary in exploration_summaries:
+        if exploration_summary is None:
+            continue
+        if exploration_summary.status == (
+                rights_manager.ACTIVITY_STATUS_PRIVATE):
+            if editor_user_id is None:
+                continue
+            if not rights_manager.Actor(editor_user_id).can_edit(
+                    feconf.ACTIVITY_TYPE_EXPLORATION,
+                    exploration_summary.id):
+                continue
+
+        filtered_exploration_summaries.append(exploration_summary)
+
+    return [{
```

Comment:
Actually, better to define a method on the summary object called `to_metadata_dict()` and use that. This helps us avoid having arbitrary dicts like this all over the codebase.